contextproject-ms3/src/main/java/contextproject/models/Key.java

```diff
@@ -0,0 +1,195 @@
+        keyCharPart = "B";
+        break;
+
+      case "C" :
+        keyNumberPart = 8;
+        keyCharPart = "B";
+        break;
+
+      case "Dd" :
+        keyNumberPart = 3;
+        keyCharPart = "B";
+        break;
+
+      case "D" :
+        keyNumberPart = 10;
+        keyCharPart = "B";
+        break;
+
+      case "Eb" :
+        keyNumberPart = 5;
+        keyCharPart = "B";
+        break;
+
+      case "E" :
+        keyNumberPart = 12;
+        keyCharPart = "B";
+        break;
+
+      case "F" :
+        keyNumberPart = 7;
+        keyCharPart = "B";
+        break;
+
+      case "Gb" :
+        keyNumberPart = 02;
+        keyCharPart = "B";
+        break;
+
+      case "G" :
+        keyNumberPart = 9;
+        keyCharPart = "B";
+        break;
+
+      case "Ab" :
+        keyNumberPart = 4;
+        keyCharPart = "B";
+        break;
+
+      case "Am" :
+        keyNumberPart = 8;
+        keyCharPart = "A";
+        break;
+
+      case "Bbm" :
+        keyNumberPart = 3;
+        keyCharPart = "A";
+        break;
+
+      case "Bm" :
+        keyNumberPart = 10;
+        keyCharPart = "A";
+        break;
+
+      case "Cm" :
+        keyNumberPart = 5;
+        keyCharPart = "A";
+        break;
+
+      case "Dbm" :
+        keyNumberPart = 12;
+        keyCharPart = "A";
+        break;
+
+      case "Dm" :
+        keyNumberPart = 7;
+        keyCharPart = "A";
+        break;
+
+      case "Ebm" :
+        keyNumberPart = 2;
+        keyCharPart = "A";
+        break;
+
+      case "Em" :
+        keyNumberPart = 9;
+        keyCharPart = "A";
+        break;
+
+      case "Fm" :
+        keyNumberPart = 4;
+        keyCharPart = "A";
+        break;
+
+      case "Gbm" :
+        keyNumberPart = 11;
+        keyCharPart = "A";
+        break;
+
+      case "Gm" :
+        keyNumberPart = 6;
+        keyCharPart = "A";
+        break;
+
+      case "Abm" :
+        keyNumberPart = 1;
+        keyCharPart = "A";
+        break;
+
+      default :
+        keyNumberPart = 0;
+        keyCharPart = null;
+        break;
+    }
+  }
+
+  /**
+   * returns the key in original music notation.
+   * 
+   * @return String
+   */
+  public String getMusicKey() {
+    return keyMusicNotiation;
+  }
+
+  /**
+   * returns the key in camelot wheel notation.
+   * 
+   * @return String
+   */
+  public String getCamelotWheelKey() {
+    return keyNumberPart + keyCharPart;
+  }
+
+  /**
+   * Get the neighbor keys using camelot wheel.
+   * 
+   * @return ArrayList
+   */
+  public ArrayList<String> getNeighborKeys() {
+    ArrayList<String> neighbors = new ArrayList<String>();
+    
+    for (int i = keyNumberPart - 1; i <= keyNumberPart + 2; i++) {
+      if (i == keyNumberPart) {
+        neighbors.add(Math.round((i % 12.5)) + "A");
+        neighbors.add(Math.round((i % 12.5)) + "B");
+      } else {
+        neighbors.add(Math.round((i % 12.5)) + keyCharPart);
+      }
+    }
+    return neighbors;
+  }
+

```

Comment:
Looking good! Could you test especially convertKey() and getNeighborKeys() according to the camelot wheel? Then I think we can merge it. :+1: